File: DSA/Krish_Mishra_590015053/Week_3/Day19_Question2.py
def find_peak(arr):
    start = 0
    end = len(arr) - 1

    while start <= end:
        mid = (start + end) // 2

        left = arr[mid - 1] if mid > 0 else float('-inf')
        right = arr[mid + 1] if mid < len(arr) - 1 else float('-inf')
        if arr[mid] >= left and arr[mid] >= right:
            return mid
        elif left > arr[mid]:
            end = mid - 1
        else:
            start = mid + 1

    return -1 

File: DSA/GEHNA_KHANNA_590013140/day19question2.c
#include <stdio.h>

int findPeak(int arr[], int n) {
    int left = 0, right = n - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        int leftOK = (mid == 0 || arr[mid] >= arr[mid - 1]);
        int rightOK = (mid == n - 1 || arr[mid] >= arr[mid + 1]);

        if (leftOK && rightOK)
            return mid;

        if (mid > 0 && arr[mid - 1] > arr[mid])
            right = mid - 1;
        else
            left = mid + 1;
    }

    return -1;
}
int main() {
    int n;

    printf("Enter the number of elements in the array: ");
    scanf("%d", &n);

    int arr[n];

    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    int peakIndex = findPeak(arr, n);
    printf("Peak element found at index: %d (value = %d)\n", peakIndex, arr[peakIndex]);

    return 0;
}



